config/config.go

```diff
@@ -101,12 +102,44 @@ const (
 	FlagLogTrace = "log.trace"
 )
 
+// DurationWrapper is a wrapper for time.Duration that implements encoding.TextMarshaler and encoding.TextUnmarshaler
+type DurationWrapper struct {
+	time.Duration
+}
+
+// MarshalText implements encoding.TextMarshaler to format the duration as text
+func (d DurationWrapper) MarshalText() ([]byte, error) {
+	return []byte(d.String()), nil
+}
+
+// UnmarshalText implements encoding.TextUnmarshaler to parse the duration from text
+func (d *DurationWrapper) UnmarshalText(text []byte) error {
+	var err error
+	d.Duration, err = time.ParseDuration(string(text))
+	return err
+}
+
+// MarshalTOML implements toml.Marshaler to format the duration as text for TOML
+func (d DurationWrapper) MarshalTOML() (interface{}, error) {
+	return d.String(), nil
+}
+
+// UnmarshalTOML implements toml.Unmarshaler to parse the duration from text in TOML
+func (d *DurationWrapper) UnmarshalTOML(v interface{}) error {
+	if s, ok := v.(string); ok {
+		var err error
+		d.Duration, err = time.ParseDuration(s)
+		return err
+	}
+	return fmt.Errorf("cannot unmarshal %T into DurationWrapper", v)
+}
+
 // Config stores Rollkit configuration.
 type Config struct {
 	// Base configuration
-	RootDir    string      `mapstructure:"home"`
-	DBPath     string      `mapstructure:"db_path"`
-	Entrypoint string      `mapstructure:"entrypoint" toml:"entrypoint"`
+	RootDir    string      `mapstructure:"home" toml:"RootDir" comment:"Root directory for all data"`
+	DBPath     string      `mapstructure:"db_path" toml:"DBPath" comment:"Path to the database directory"`
+	Entrypoint string      `mapstructure:"entrypoint" toml:"entrypoint" comment:"Path to the rollup entrypoint (main.go)"`
```

Comment:
This reads odd. Why does the code need to know where main.go is?

Reply:
this is because rollkit was designet to be a wrapper around the project. You execute rollkit and it build the entrypoint which needs to have access to the main.go. It builds it as binary and redirects the flags.